examples/vision/deeplabv3_plus.py

```diff
@@ -0,0 +1,298 @@
+relying on hand-crafted features on benchmark tasks. In this example, we would implement
+the **DeepLabV3+** model as proposed by the paper [Encoder-Decoder with Atrous Separable
+Convolution for Semantic Image Segmentation](https://arxiv.org/pdf/1802.02611.pdf) for
+multi-class semantic segmentation.
+"""
+
+"""
+## Download Dataset
+"""
+
+import os
+import cv2
+import numpy as np
+from glob import glob
+import tensorflow as tf
+from scipy.io import loadmat
+import matplotlib.pyplot as plt
+
+"""shell
+!gdown https://drive.google.com/uc?id=1B9A9UCJYMwTL4oBEo4RZfbMZMaZhKJaz
+!unzip -q instance-level-human-parsing.zip
+"""
+
+"""
+
+"""
+
+"""
+## Building Tensorflow Dataset
+"""
+
+"""
+For this example, we would be using 200 images from the instance-level human parsing
+dataset to train our model.
+"""
+
+IMAGE_SIZE = 512
+BATCH_SIZE = 4
+NUM_CLASSES = 20
+DATA_DIR = "./instance-level_human_parsing/instance-level_human_parsing/Training"
+MAX_IMAGES = 200
+
+train_images = sorted(glob(os.path.join(DATA_DIR, "Images/*")))[:MAX_IMAGES]
+train_masks = sorted(glob(os.path.join(DATA_DIR, "Category_ids/*")))[:MAX_IMAGES]
+
+
+def read_image(image_path, mask=False):
+    image = tf.io.read_file(image_path)
+    if mask:
+        image = tf.image.decode_png(image, channels=1)
+        image.set_shape([None, None, 1])
+        image = tf.image.resize(images=image, size=[IMAGE_SIZE, IMAGE_SIZE])
+        image = tf.cast(image, tf.float32)
+    else:
+        image = tf.image.decode_png(image, channels=3)
+        image.set_shape([None, None, 3])
+        image = tf.image.resize(images=image, size=[IMAGE_SIZE, IMAGE_SIZE])
+        image = tf.cast(image, tf.float32) / 127.5 - 1
+    return image
+
+
+def load_data(image_list, mask_list):
+    image = read_image(image_list)
+    mask = read_image(mask_list, mask=True)
+    return image, mask
+
+
+def data_generator(image_list, mask_list):
+    dataset = tf.data.Dataset.from_tensor_slices((image_list, mask_list))
+    dataset = dataset.map(load_data, num_parallel_calls=tf.data.AUTOTUNE)
+    dataset = dataset.batch(BATCH_SIZE, drop_remainder=True)
+    dataset = dataset.repeat(1)
+    dataset = dataset.prefetch(tf.data.AUTOTUNE)
+    return dataset
+
+
+dataset = data_generator(train_images, train_masks)
+dataset
+
+"""
+## Building DeepLabV3+ Model
+"""
+
+"""
+DeepLabv3+ extends DeepLabv3 by employing a encoder-decoder structure. The encoder module
+encodes multi-scale contextual information by applying atrous convolution at multiple
+scales, while the simple yet effective decoder module refines the segmentation results
+along object boundaries.
+
+![](https://github.com/lattice-ai/DeepLabV3-Plus/raw/master/assets/deeplabv3_plus_diagram.png)
+![](https://github.com/lattice-ai/DeepLabV3-Plus/raw/master/assets/deeplabv3_plus_diagram.png)
+
+
+"""
+
+"""
+**Atrous Convolution:** With Atrous convolution, as we go deeper, we can keep the stride
```

Comment:
It's called "dilated convolution". "Convolution a trous" (literally "sparse convolution") is a misnomer.

Reply:
Changed to dilated convolution.